src/static/js/getChatUpdate.js

```diff
@@ -93,11 +100,59 @@ document
   .getElementById("messageContainer")
   .addEventListener("click", function (event) {
     if (event.target.classList.contains("message")) {
-      event.target.style.backgroundColor = "#e0e0e0";
-      console.log("Element clicked:", event.target.textContent);
+
+      document.querySelectorAll(".delete-button").forEach((button) => {
+        button.parentNode.style.backgroundColor = "white";
+        button.remove();
+      });
+
+      event.target.parentNode.style.backgroundColor = "#e0e0e0";
+
+      var deleteButton = document.createElement("button");
+      deleteButton.textContent = "Delete";
+      deleteButton.classList.add("delete-button");
+      deleteButton.style.marginLeft = "10px";
+
+      deleteButton.addEventListener("click", function () {
+        const tokenId = event.target.parentNode.id;
+        deleteMessage(tokenId);
+        deleteButton.parentNode.style.backgroundColor = "white";
+        deleteButton.remove();
+      });
+
+      event.target.parentNode.appendChild(deleteButton);
+
     }
   });
 
+function deleteMessage(tokenId){
+  const email = document.getElementById("email").value;
+  const tID = document.getElementById("tID").value;
+  const csrfToken = document.getElementsByName('csrfmiddlewaretoken')[0].value;
+
+  if (message != "") {
```

Comment:
Thanks for pointing that out! I was planning to make the message unclickable once it is deleted. My concern about checking if updates are only happening once is that this will query the whole database on all messages for that chat in the backend for every delete request, which makes a little bit worried about time efficiency as the chat history grows.

Maybe in future implementations, we will prioritize in disabling resending delete requests? I think that might be a more time efficient solutions for this?

Reply:
Okay that makes sense to me for now as long as the message is unclickable upon deletion!